xl2times/transforms.py

```diff
@@ -822,20 +820,19 @@ def complete_dictionary(
     tables: Dict[str, DataFrame],
     model: datatypes.TimesModel,
 ) -> Dict[str, DataFrame]:
-
-    for k, v in {
-        "AllRegions": model.all_regions,
-        "Regions": model.internal_regions,
-        "DataYears": model.data_years,
-        "PastYears": model.past_years,
-        "ModelYears": model.model_years,
-    }.items():
+    for k, v in [
```

Comment:
Just curious as to why it is better to use a list tuples here instead of a dictionary. Should the dictionary below be converted as well?

Reply:
Ah, this is because iterating through a set results in a non-deterministic order (see also #50 and #67). This makes it very difficult to debug regressions, because I rely on the `--verbose` flag and a diff tool to find out which transformation caused the regression. But with nondeterministic behaviour, there are too many changes in the diff..

I'm considering adding a check to CI that runs the tool on a small benchmark ~5-10 times and ensures that all intermediate tables are identical across runs.

Reply:
I left the dictionary below unchanged because the loop only assigns to a dictionary, so the order doesn't matter.